### PR TITLE
drivers/ft5x06: allow multiple device types

### PR DESCRIPTION
**Pull request by leandrolanzieri**
### Contribution description
As stated in #17448 the ft5x06 driver has not changes between models, other than vendor ID, needed calibration and maximum number of touches. Although this is handled via macros, if more than one variant is used at the same time the initialization would fail if devices have different vendor ID. Similar issue would occur with the maximum number of touches.

Taking advantage of the driver being generic enough, this PR proposes to add a param to specify the device model, similar to what's done for the `sx126x`, to allow more than one device of different variants to work at the same time.

### Testing procedure
- Check that the driver works for both supported boards

### Issues/PRs references
#17448

### CODE REVIEW

**Comment by leandrolanzieri** [comment]
I was actually hoping we could get the device ID information during initialization instead of adding it to params, but I could not find the actual values.

> I agree with the idea proposed here, even if I didn't do it that way in the first place (since other drivers have the same limitation). But if we go this way, then there's no need for pseudo-modules anymore so more things must be changed (in Makefile.dep, Kconfig, in boards Makefile.dep and the test application Makefile).

I removed the unnecessary pseudomodules.

File: drivers/ft5x06/ft5x06_touch_dev.c [hunk 1]
```diff
@@ -55,8 +55,16 @@ uint8_t _ft5x06_touches(const touch_dev_t *touch_dev, touch_t *touches, size_t l
...
     ft5x06_read_touch_count(dev, &ret);
 
+    uint8_t count_max;
+    if (dev->params.type == FT5X06_TYPE_FT6X06 || dev->params.type == FT5X06_TYPE_FT6X36) {
+        count_max = FT6XX6_TOUCHES_COUNT_MAX;
+    }
+    else {
+        count_max = FT5X06_TOUCHES_COUNT_MAX;
+    }
+
```
Comment (aabadie) [inline drivers/ft5x06/ft5x06_touch_dev.c:65]: Maybe add an internal function to get the number of max touches and avoid a duplicate code.

File: drivers/include/ft5x06.h [hunk 2]
```diff
@@ -71,11 +88,12 @@ typedef void (*ft5x06_event_cb_t)(void *arg);
  * @brief   Device initialization parameters
  */
 typedef struct {
-    i2c_t i2c;              /**< I2C device which is used */
-    uint8_t addr;           /**< Device I2C address */
-    gpio_t int_pin;         /**< Touch screen interrupt pin */
-    uint16_t xmax;          /**< Touch screen max X position */
-    uint16_t ymax;          /**< Touch screen max Y position */
```
Comment (aabadie) [inline drivers/include/ft5x06.h:78]: I don't think the doxygen comments should be shifted. This would only result in a one line being added.

Comment (leandrolanzieri) [reply drivers/include/ft5x06.h]: Reverted in 06bd29c67dc805e2ba924bc208c5ed04284e08f6

File: drivers/ft5x06/ft5x06_touch_dev.c (hunk 1 above)
Comment (leandrolanzieri) [reply drivers/ft5x06/ft5x06_touch_dev.c]: Factored out in 7d49dec8ced116282e798748cbc7e816fbec4738

File: drivers/include/ft5x06.h [hunk 3]
```diff
@@ -71,11 +88,12 @@ typedef void (*ft5x06_event_cb_t)(void *arg);
  * @brief   Device initialization parameters
  */
 typedef struct {
-    i2c_t i2c;              /**< I2C device which is used */
-    uint8_t addr;           /**< Device I2C address */
-    gpio_t int_pin;         /**< Touch screen interrupt pin */
+    i2c_t i2c;               /**< I2C device which is used */
+    uint8_t addr;            /**< Device I2C address */
+    gpio_t int_pin;          /**< Touch screen interrupt pin */
```
Comment (aabadie) [inline drivers/include/ft5x06.h:93]: Still shifted by one space :)

File: drivers/ft5x06/include/ft5x06_params.h [hunk 4]
```diff
@@ -49,13 +49,17 @@ extern "C" {
 #ifndef FT5X06_PARAM_YMAX
 #define FT5X06_PARAM_YMAX               (272U)
 #endif
+#ifndef FT5X06_PARAM_TYPE
+#define FT5X06_PARAM_TYPE               FT5X06_TYPE_FT5336
```
Comment (aabadie) [inline drivers/ft5x06/include/ft5x06_params.h:53]: This is missing documentation. Other defines are added in the [doccheck excluded patterns list](https://github.com/RIOT-OS/RIOT/blob/master/dist/tools/doccheck/exclude_patterns).

Comment (leandrolanzieri) [reply drivers/ft5x06/include/ft5x06_params.h:53]: Thanks, added in cd30be5b8b40cb0dcfee37297e26682ef70747a8

File: drivers/include/ft5x06.h (hunk 3 above)
Comment (leandrolanzieri) [reply drivers/include/ft5x06.h]: 😅  f6fd2ffde3f2c4aa696ab0cda4918d81758a492c

**Comment by aabadie** [comment]
Please squash!

**Comment by leandrolanzieri** [comment]
> Please squash!

Squashed!

**Comment by aabadie** [comment]
Seems like Murdock found an issue.

**Comment by leandrolanzieri** [comment]
> Seems like Murdock found an issue.

I forgot to update the Kconfig `HAVE_` symbols from both boards

**Pull request merged**